File: src/model/rules/gameRule/AbstractGameRuleMonochrome.java
package model.rules.gameRule;

import model.enums.GameType;
import model.pieces.PieceImplMonochrome;

public abstract class AbstractGameRuleMonochrome implements GameRule {

    @Override
    public GameType getGameType() {
        return GameType.PLACE_PIECE;
    }

    public PieceImplMonochrome[][] basicInitializeGrid(int height, int width){
        PieceImplMonochrome[][] pieceGrid = new PieceImplMonochrome[height+2][width+2];
        for(int i = 0; i < height+2; i++) {
            for(int j = 0; j < width+2; j++) {
                pieceGrid[i][j] = new PieceImplMonochrome();
            }
        }
        return pieceGrid;
    }
}

File: src/model/rules/gameRule/AbstractGameRuleMonochrome.kt
package model.rules.gameRule

import model.enums.GameType
import model.pieces.Piece
import model.pieces.PieceImplBomb
import model.pieces.PieceImplMonochrome

abstract class AbstractGameRuleMonochrome : GameRule {
    override val gameType: GameType
        get() = GameType.PLACE_PIECE

    fun basicInitializeGrid(height: Int, width: Int): Array<Array<Piece>> {
        val pieceGrid: Array<Array<Piece>> = Array(height + 2) {
            Array(width + 2) { PieceImplMonochrome() }
        }
        return pieceGrid
    }
}
